File: rb/features.rb
# Dictum SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module DictumFeatures
  def self.make_feature(name)
    case name
    when "base"
      DictumBaseFeature.new
    when "ratelimit"
      DictumRatelimitFeature.new
    when "retry"
      DictumRetryFeature.new
    when "test"
      DictumTestFeature.new
    when "timeout"
      DictumTimeoutFeature.new
    else
      DictumBaseFeature.new
    end
  end
end

File: py/features.py
# Dictum SDK feature factory

from feature.base_feature import DictumBaseFeature
from feature.test_feature import DictumTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DictumBaseFeature(),
        "test": lambda: DictumTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
